File: zad3_tripleList_python/Node.py

class Node:

    def __init__(self, val):
        self.value = val
        self.ancestor = None
        self.coeval = None
        self.offspring = None

    def __init__(self, val, prev):
        self.value = val
        self.ancestor = prev
        self.coeval = None
        self.offspring = None

    def set_previous(self, node):
        self.ancestor = node

    def set_middle(self, node):
        self.coeval = node

    def set_next(self, node):
        self.offspring = node

    def set_value(self, val):
        self.value = val

    def get_previous(self):
        return self.ancestor

    def get_middle(self):
        return self.coeval

    def get_next(self):
        return self.offspring

    def get_value(self):
        return self.value

File: zad3_trippleList_java/src/Node.java
/**
 * Created by Makda on 2015-12-29.
 */
public class Node <T> {
    T value;
    Node ancestor;
    Node offspring;
    Node coeval;

    public Node(T value){
        this.value = value;
        this.offspring = null;
        this.ancestor = null;
        this.coeval = null;
    }

    public Node(T value, Node ancestor){
        this.value = value;
        this.offspring = null;
        this.ancestor = ancestor;
        this.coeval = null;
    }

    public T getValue() {
        return value;
    }

    public void setValue(T value) {
        this.value = value;
    }

    public Node getAncestor() {
        return ancestor;
    }

    public void setAncestor(Node ancestor) {
        this.ancestor = ancestor;
    }

    public Node getOffspring() {
        return offspring;
    }

    public void setOffspring(Node offspring) {
        this.offspring = offspring;
    }

    public Node getCoeval() {
        return coeval;
    }

    public void setCoeval(Node coeval) {
        this.coeval = coeval;
    }
}
